core/src/main/java/com/onelogin/saml2/util/Util.java

```diff
@@ -1356,14 +1354,13 @@ public static String generateUniqueID() {
 			MessageDigest crypt = MessageDigest.getInstance("SHA-1");
 			crypt.reset();
 			crypt.update(id.getBytes());
-			uniqueIdSha1 = new BigInteger(1, crypt.digest()).toString(16);
+			final String uniqueIdSha1 = new BigInteger(1, crypt.digest()).toString(16);
```

Comment:
@pitbulk any reason the ids are generated this way, rather than a simple `return UNIQUE_ID_PREFIX + UUID.randomUUID()`?

This seems a bit overcomplicated, uses `java.util.Random` which is not a cryptographically secure random number generator. It does append a random UUID, which does use SecureRandom, however I was wondering whether all of this is needed (the Random prefix, and the SHA-1 hashing).